arangod/Agency/State.cpp

```diff
@@ -788,8 +790,23 @@ bool State::checkCollection(std::string const& name) {
   return true;
 }
 
+/// Drop
+void State::dropCollection(std::string const& colName) {
+  try {
+    auto col = _vocbase->lookupCollection(colName)->id();
```

Comment:
This is not safe, as `lookupCollection` will return a nullptr in case the collection does not exist.

Reply:
Fixed.